File: source/downloads/code/LeetCode/Java/Simplify-path.java
public class Solution {
    public String simplifyPath(String path) {
        Deque<String> stack = new LinkedList<String>();
 
        for (String s : path.split("/")){
            if(s.equals(".") || s.equals(""))
                continue;
            if(s.equals("..")){
                if(!stack.isEmpty())
                    stack.pop();
                continue;
            }
            
            stack.push(s);

        }
    // 生成字符串时要从stack的底部开始       
    String res = "";
    for (String dir : stack) res = "/" + dir + res;
    return res.isEmpty() ? "/" : res;
    }
}

File: downloads/code/LeetCode/Python/Simplify-Path.py
class Solution(object):
    def simplifyPath(self, path):
        """
        :type path: str
        :rtype: str
        """
        ret = ""
        if path == None or len(path) == 0:
            return 
        paths = path.split('/')
        stack = []
        for i in paths:
            if i == "..":
                if len(stack) != 0:
                    stack.pop()
            elif i == "." or i =="":
                continue
            else:
                stack.append(i)
        ret = '/'+'/'.join(stack)
        return ret
    
